### PR TITLE
(get|set)-route: add user_created flag

### PR DESCRIPTION
**Pull request by Amygos**
Using `user_created` flag is possible to mark a route as manually created by a user, in this way the UI can avoid a system route begin deleted or modified by mistake.

### CODE REVIEW

**Review by andre8244 — APPROVED** [review]

File: imageroot/pypkg/get_route.py [hunk 1]
```diff
@@ -35,6 +35,10 @@ def get_route(data):
     route = {}
     api_path = os.environ["API_PATH"]
 
+    agent_id = os.getenv("AGENT_ID", "")
+    if not agent_id:
+        raise Exception("AGENT_ID not found inside the environemnt")
+
```
Comment (DavidePrincipi) [inline imageroot/pypkg/get_route.py:41]: IIUC the "Pythonic way" it's easier, more readable and "debuggable" to (1) not provide a default and check the return value, (2) let the exception raise (3) handle it somewhere, eventually. 

In this case it's unlikely that AGENT_ID is not set in the environment: that would be a major issue. If that happens anyway the default handler is enough for us. 

The error message and the stack trace would help to find the problem.

```suggestion
    agent_id = os.environ["AGENT_ID"]
```

File: imageroot/pypkg/get_route.py [hunk 2]
```diff
@@ -79,6 +83,10 @@ def get_route(data):
         if route.get("path"):
             route['strip_prefix'] = True if middlewares and f'{module}-stripprefix@redis' in middlewares else False
 
+        # Check if the route was created manually
+        r = agent.redis_connect(privileged=True)
+        route['user_created'] = r.sismember(f'{agent_id}/user_created_routes', data["instance"])
```
Comment (DavidePrincipi) [inline imageroot/pypkg/get_route.py:88]: Single-char variable names are hard to search in the code. :smile_cat: please type more chars!

```suggestion
        rdb = agent.redis_connect(privileged=True)
        route['user_created'] = rdb.sismember(f'{agent_id}/user_created_routes', data["instance"])
```

Furthermore variable `r` is already used in the scope for multiple regexp match assignments. ..What about renaming it too and using more variables? e.g. `host_match`, `path_match`...

**Review by DavidePrincipi — CHANGES_REQUESTED** [review]

File: imageroot/pypkg/get_route.py (hunk 2 above)
Comment (Amygos) [reply imageroot/pypkg/get_route.py]: I will change the variables' names outside of this PR.

**Review by DavidePrincipi — APPROVED** [review]
Ok to refactor variable names in another PR

**Pull request merged**